cpp/src/parquet/schema.cc

```diff
@@ -409,6 +409,32 @@ std::unique_ptr<Node> GroupNode::FromParquet(const void* opaque_element, int nod
   return std::unique_ptr<Node>(group_node.release());
 }
 
+namespace {
+
+// If the parquet file is corrupted it is possible the type value decoded
+// will not be in the range of format::Type::type, which is undefined behavior.
+// This method prevents this by loading the value as the underlying type and checking
+// to make sure it is in range.
+template <typename ApiType, typename ThriftType>
+inline typename ApiType::type SafeLoad(ThriftType* in) {
+  using ApiTypeEnum = typename ApiType::type;
+  using ApiTypeRawEnum = typename std::underlying_type<ApiTypeEnum>::type;
+  static_assert(std::is_unsigned<ApiTypeRawEnum>::value,
```

Comment:
`ApiTypeRawEnum::value`?

Reply:
Not sure exactly what you are asking.
ApiTypeRawEnum is a type like uint32_t. I believe is_unsigned determines if the type is unsigned by accessing the static member value on the produced struct (https://en.cppreference.com/w/cpp/types/is_unsigned)

Reply:
Ah, sorry, I misread.